### PR TITLE
bug/65881 SPA crashes without app insights configured

### PR DESCRIPTION
**Pull request by GuyHarwood**
missing connection string for app insights always initialises to empty string

### CODE REVIEW

**Review by jon-shipley — APPROVED** [review]
lgtm 👍

File: pupil-spa/src/app/string-util.spec.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  })
+
+  it('should return true for empty string', () => {
+    expect(stringUtil.isEmptyOrUndefined('')).toBe(true)
+  })
+
+  it('should return false for non-empty string', () => {
+    expect(stringUtil.isEmptyOrUndefined('a')).toBe(false)
+  })
+})
```
Comment (jon-shipley) [inline pupil-spa/src/app/string-util.spec.ts:17]: No tests for bottom values?

**Review by jon-shipley — APPROVED** [review]
Lgtm 👍

Nice to see a string utility helper, but it could use a test for undefined.

File: pupil-spa/src/app/string-util.spec.ts (hunk 1 above)
Comment (GuyHarwood) [reply pupil-spa/src/app/string-util.spec.ts:17]: line 7 tests for undefined...

Comment (jon-shipley) [reply pupil-spa/src/app/string-util.spec.ts:17]: Great!

**Pull request merged**